File: symbol/dimgui_io.ts
// deno-fmt-ignore-file
const ImVec2 = ["f32","f32"] as const;
const dimguiIOSymbols = {
    DImGuiIOGetConfigFlags:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigFlags:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetBackendFlags:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetBackendFlags:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetDisplaySize:{
        parameters: ["pointer"],
        result: {struct:ImVec2},
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetDisplaySize:{
        parameters: ["pointer",{struct:ImVec2}],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetDeltaTime:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetDeltaTime:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetIniSavingRate:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetIniSavingRate:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetIniFilename:{
        parameters: ["pointer"],
        result: "buffer",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetIniFilename:{
        parameters: ["pointer","buffer"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetLogFilename:{
        parameters: ["pointer"],
        result: "buffer",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetLogFilename:{
        parameters: ["pointer","buffer"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMouseDoubleClickTime:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMouseDoubleClickTime:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMouseDoubleClickMaxDist:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMouseDoubleClickMaxDist:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMouseDragThreshold:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMouseDragThreshold:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetKeyRepeatDelay:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetKeyRepeatDelay:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetKeyRepeatRate:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetKeyRepeatRate:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetHoverDelayNormal:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetHoverDelayNormal:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetHoverDelayShort:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetHoverDelayShort:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetUserData:{
        parameters: ["pointer"],
        result: "buffer",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetUserData:{
        parameters: ["pointer","buffer"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetFonts:{
        parameters: ["pointer"],
        result: "pointer",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetFonts:{
        parameters: ["pointer","pointer"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetFontGlobalScale:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetFontGlobalScale:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetFontAllowUserScaling:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetFontAllowUserScaling:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetFontDefault:{
        parameters: ["pointer"],
        result: "pointer",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetFontDefault:{
        parameters: ["pointer","pointer"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetDisplayFramebufferScale:{
        parameters: ["pointer"],
        result: {struct:ImVec2},
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetDisplayFramebufferScale:{
        parameters: ["pointer",{struct:ImVec2}],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigDockingNoSplit:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigDockingNoSplit:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigDockingWithShift:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigDockingWithShift:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigDockingAlwaysTabBar:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigDockingAlwaysTabBar:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigDockingTransparentPayload:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigDockingTransparentPayload:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigViewportsNoAutoMerge:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigViewportsNoAutoMerge:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigViewportsNoTaskBarIcon:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigViewportsNoTaskBarIcon:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigViewportsNoDecoration:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigViewportsNoDecoration:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigViewportsNoDefaultParent:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigViewportsNoDefaultParent:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMouseDrawCursor:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMouseDrawCursor:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigMacOSXBehaviors:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigMacOSXBehaviors:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigInputTrickleEventQueue:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigInputTrickleEventQueue:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigInputTextCursorBlink:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigInputTextCursorBlink:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigInputTextEnterKeepActive:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigInputTextEnterKeepActive:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigDragClickToInputText:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigDragClickToInputText:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigWindowsResizeFromEdges:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigWindowsResizeFromEdges:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigWindowsMoveFromTitleBarOnly:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigWindowsMoveFromTitleBarOnly:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetConfigMemoryCompactTimer:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetConfigMemoryCompactTimer:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetWantCaptureMouse:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetWantCaptureMouse:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetWantCaptureKeyboard:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetWantCaptureKeyboard:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetWantTextInput:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetWantTextInput:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetWantSetMousePos:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetWantSetMousePos:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetWantSaveIniSettings:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetWantSaveIniSettings:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetNavActive:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetNavActive:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetNavVisible:{
        parameters: ["pointer"],
        result: "bool",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetNavVisible:{
        parameters: ["pointer","bool"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetFramerate:{
        parameters: ["pointer"],
        result: "f32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetFramerate:{
        parameters: ["pointer","f32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMetricsRenderVertices:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMetricsRenderVertices:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMetricsRenderIndices:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMetricsRenderIndices:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMetricsRenderWindows:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMetricsRenderWindows:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMetricsActiveWindows:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMetricsActiveWindows:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMetricsActiveAllocations:{
        parameters: ["pointer"],
        result: "i32",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMetricsActiveAllocations:{
        parameters: ["pointer","i32"],
        result: "void",
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOGetMouseDelta:{
        parameters: ["pointer"],
        result: {struct:ImVec2},
    } as const satisfies Deno.ForeignFunction,
    DImGuiIOSetMouseDelta:{
        parameters: ["pointer",{struct:ImVec2}],
        result: "void",
    } as const satisfies Deno.ForeignFunction,

} as const satisfies Deno.ForeignLibraryInterface;
export default dimguiIOSymbols

File: imgui/src/dimgui_io.cpp
#include "../dim_config.h"

// use sturcts without constructors
#define CIMGUI_DEFINE_ENUMS_AND_STRUCTS
#include "cimgui/cimgui.h"

// clang-format off
DIMGUI_EXPORT ImGuiConfigFlags DImGuiIOGetConfigFlags (ImGuiIO* s){ return s->ConfigFlags;}
DIMGUI_EXPORT ImGuiBackendFlags DImGuiIOGetBackendFlags (ImGuiIO* s){ return s->BackendFlags;}
DIMGUI_EXPORT ImVec2 DImGuiIOGetDisplaySize (ImGuiIO* s){ return s->DisplaySize;}
DIMGUI_EXPORT float DImGuiIOGetDeltaTime (ImGuiIO* s){ return s->DeltaTime;}
DIMGUI_EXPORT float DImGuiIOGetIniSavingRate (ImGuiIO* s){ return s->IniSavingRate;}
DIMGUI_EXPORT const char* DImGuiIOGetIniFilename (ImGuiIO* s){ return s->IniFilename;}
DIMGUI_EXPORT const char* DImGuiIOGetLogFilename (ImGuiIO* s){ return s->LogFilename;}
DIMGUI_EXPORT float DImGuiIOGetMouseDoubleClickTime (ImGuiIO* s){ return s->MouseDoubleClickTime;}
DIMGUI_EXPORT float DImGuiIOGetMouseDoubleClickMaxDist (ImGuiIO* s){ return s->MouseDoubleClickMaxDist;}
DIMGUI_EXPORT float DImGuiIOGetMouseDragThreshold (ImGuiIO* s){ return s->MouseDragThreshold;}
DIMGUI_EXPORT float DImGuiIOGetKeyRepeatDelay (ImGuiIO* s){ return s->KeyRepeatDelay;}
DIMGUI_EXPORT float DImGuiIOGetKeyRepeatRate (ImGuiIO* s){ return s->KeyRepeatRate;}
DIMGUI_EXPORT float DImGuiIOGetHoverDelayNormal (ImGuiIO* s){ return s->HoverDelayNormal;}
DIMGUI_EXPORT float DImGuiIOGetHoverDelayShort (ImGuiIO* s){ return s->HoverDelayShort;}
DIMGUI_EXPORT void* DImGuiIOGetUserData (ImGuiIO* s){ return s->UserData;}
DIMGUI_EXPORT ImFontAtlas* DImGuiIOGetFonts (ImGuiIO* s){ return s->Fonts;}
DIMGUI_EXPORT float DImGuiIOGetFontGlobalScale (ImGuiIO* s){ return s->FontGlobalScale;}
DIMGUI_EXPORT bool DImGuiIOGetFontAllowUserScaling (ImGuiIO* s){ return s->FontAllowUserScaling;}
DIMGUI_EXPORT ImFont* DImGuiIOGetFontDefault (ImGuiIO* s){ return s->FontDefault;}
DIMGUI_EXPORT ImVec2 DImGuiIOGetDisplayFramebufferScale (ImGuiIO* s){ return s->DisplayFramebufferScale;}
DIMGUI_EXPORT bool DImGuiIOGetConfigDockingNoSplit (ImGuiIO* s){ return s->ConfigDockingNoSplit;}
DIMGUI_EXPORT bool DImGuiIOGetConfigDockingWithShift (ImGuiIO* s){ return s->ConfigDockingWithShift;}
DIMGUI_EXPORT bool DImGuiIOGetConfigDockingAlwaysTabBar (ImGuiIO* s){ return s->ConfigDockingAlwaysTabBar;}
DIMGUI_EXPORT bool DImGuiIOGetConfigDockingTransparentPayload (ImGuiIO* s){ return s->ConfigDockingTransparentPayload;}
DIMGUI_EXPORT bool DImGuiIOGetConfigViewportsNoAutoMerge (ImGuiIO* s){ return s->ConfigViewportsNoAutoMerge;}
DIMGUI_EXPORT bool DImGuiIOGetConfigViewportsNoTaskBarIcon (ImGuiIO* s){ return s->ConfigViewportsNoTaskBarIcon;}
DIMGUI_EXPORT bool DImGuiIOGetConfigViewportsNoDecoration (ImGuiIO* s){ return s->ConfigViewportsNoDecoration;}
DIMGUI_EXPORT bool DImGuiIOGetConfigViewportsNoDefaultParent (ImGuiIO* s){ return s->ConfigViewportsNoDefaultParent;}
DIMGUI_EXPORT bool DImGuiIOGetMouseDrawCursor (ImGuiIO* s){ return s->MouseDrawCursor;}
DIMGUI_EXPORT bool DImGuiIOGetConfigMacOSXBehaviors (ImGuiIO* s){ return s->ConfigMacOSXBehaviors;}
DIMGUI_EXPORT bool DImGuiIOGetConfigInputTrickleEventQueue (ImGuiIO* s){ return s->ConfigInputTrickleEventQueue;}
DIMGUI_EXPORT bool DImGuiIOGetConfigInputTextCursorBlink (ImGuiIO* s){ return s->ConfigInputTextCursorBlink;}
DIMGUI_EXPORT bool DImGuiIOGetConfigInputTextEnterKeepActive (ImGuiIO* s){ return s->ConfigInputTextEnterKeepActive;}
DIMGUI_EXPORT bool DImGuiIOGetConfigDragClickToInputText (ImGuiIO* s){ return s->ConfigDragClickToInputText;}
DIMGUI_EXPORT bool DImGuiIOGetConfigWindowsResizeFromEdges (ImGuiIO* s){ return s->ConfigWindowsResizeFromEdges;}
DIMGUI_EXPORT bool DImGuiIOGetConfigWindowsMoveFromTitleBarOnly (ImGuiIO* s){ return s->ConfigWindowsMoveFromTitleBarOnly;}
DIMGUI_EXPORT float DImGuiIOGetConfigMemoryCompactTimer (ImGuiIO* s){ return s->ConfigMemoryCompactTimer;}
DIMGUI_EXPORT bool DImGuiIOGetWantCaptureMouse (ImGuiIO* s){ return s->WantCaptureMouse;}
DIMGUI_EXPORT bool DImGuiIOGetWantCaptureKeyboard (ImGuiIO* s){ return s->WantCaptureKeyboard;}
DIMGUI_EXPORT bool DImGuiIOGetWantTextInput (ImGuiIO* s){ return s->WantTextInput;}
DIMGUI_EXPORT bool DImGuiIOGetWantSetMousePos (ImGuiIO* s){ return s->WantSetMousePos;}
DIMGUI_EXPORT bool DImGuiIOGetWantSaveIniSettings (ImGuiIO* s){ return s->WantSaveIniSettings;}
DIMGUI_EXPORT bool DImGuiIOGetNavActive (ImGuiIO* s){ return s->NavActive;}
DIMGUI_EXPORT bool DImGuiIOGetNavVisible (ImGuiIO* s){ return s->NavVisible;}
DIMGUI_EXPORT float DImGuiIOGetFramerate (ImGuiIO* s){ return s->Framerate;}
DIMGUI_EXPORT int DImGuiIOGetMetricsRenderVertices (ImGuiIO* s){ return s->MetricsRenderVertices;}
DIMGUI_EXPORT int DImGuiIOGetMetricsRenderIndices (ImGuiIO* s){ return s->MetricsRenderIndices;}
DIMGUI_EXPORT int DImGuiIOGetMetricsRenderWindows (ImGuiIO* s){ return s->MetricsRenderWindows;}
DIMGUI_EXPORT int DImGuiIOGetMetricsActiveWindows (ImGuiIO* s){ return s->MetricsActiveWindows;}
DIMGUI_EXPORT int DImGuiIOGetMetricsActiveAllocations (ImGuiIO* s){ return s->MetricsActiveAllocations;}
DIMGUI_EXPORT ImVec2 DImGuiIOGetMouseDelta (ImGuiIO* s){ return s->MouseDelta;}




DIMGUI_EXPORT void DImGuiIOSetConfigFlags (ImGuiIO* s, ImGuiConfigFlags value){ s->ConfigFlags=value;}
DIMGUI_EXPORT void DImGuiIOSetBackendFlags (ImGuiIO* s, ImGuiBackendFlags value){ s->BackendFlags=value;}
DIMGUI_EXPORT void DImGuiIOSetDisplaySize (ImGuiIO* s, ImVec2 value){ s->DisplaySize=value;}
DIMGUI_EXPORT void DImGuiIOSetDeltaTime (ImGuiIO* s, float value){ s->DeltaTime=value;}
DIMGUI_EXPORT void DImGuiIOSetIniSavingRate (ImGuiIO* s, float value){ s->IniSavingRate=value;}
DIMGUI_EXPORT void DImGuiIOSetIniFilename (ImGuiIO* s, char* value){ s->IniFilename=value;}
DIMGUI_EXPORT void DImGuiIOSetLogFilename (ImGuiIO* s, char* value){ s->LogFilename=value;}
DIMGUI_EXPORT void DImGuiIOSetMouseDoubleClickTime (ImGuiIO* s, float value){ s->MouseDoubleClickTime=value;}
DIMGUI_EXPORT void DImGuiIOSetMouseDoubleClickMaxDist (ImGuiIO* s, float value){ s->MouseDoubleClickMaxDist=value;}
DIMGUI_EXPORT void DImGuiIOSetMouseDragThreshold (ImGuiIO* s, float value){ s->MouseDragThreshold=value;}
DIMGUI_EXPORT void DImGuiIOSetKeyRepeatDelay (ImGuiIO* s, float value){ s->KeyRepeatDelay=value;}
DIMGUI_EXPORT void DImGuiIOSetKeyRepeatRate (ImGuiIO* s, float value){ s->KeyRepeatRate=value;}
DIMGUI_EXPORT void DImGuiIOSetHoverDelayNormal (ImGuiIO* s, float value){ s->HoverDelayNormal=value;}
DIMGUI_EXPORT void DImGuiIOSetHoverDelayShort (ImGuiIO* s, float value){ s->HoverDelayShort=value;}
DIMGUI_EXPORT void DImGuiIOSetUserData (ImGuiIO* s, void* value){ s->UserData=value;}
DIMGUI_EXPORT void DImGuiIOSetFonts (ImGuiIO* s, ImFontAtlas* value){ s->Fonts=value;}
DIMGUI_EXPORT void DImGuiIOSetFontGlobalScale (ImGuiIO* s, float value){ s->FontGlobalScale=value;}
DIMGUI_EXPORT void DImGuiIOSetFontAllowUserScaling (ImGuiIO* s, bool value){ s->FontAllowUserScaling=value;}
DIMGUI_EXPORT void DImGuiIOSetFontDefault (ImGuiIO* s, ImFont* value){ s->FontDefault=value;}
DIMGUI_EXPORT void DImGuiIOSetDisplayFramebufferScale (ImGuiIO* s, ImVec2 value){ s->DisplayFramebufferScale=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigDockingNoSplit (ImGuiIO* s, bool value){ s->ConfigDockingNoSplit=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigDockingWithShift (ImGuiIO* s, bool value){ s->ConfigDockingWithShift=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigDockingAlwaysTabBar (ImGuiIO* s, bool value){ s->ConfigDockingAlwaysTabBar=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigDockingTransparentPayload (ImGuiIO* s, bool value){ s->ConfigDockingTransparentPayload=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigViewportsNoAutoMerge (ImGuiIO* s, bool value){ s->ConfigViewportsNoAutoMerge=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigViewportsNoTaskBarIcon (ImGuiIO* s, bool value){ s->ConfigViewportsNoTaskBarIcon=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigViewportsNoDecoration (ImGuiIO* s, bool value){ s->ConfigViewportsNoDecoration=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigViewportsNoDefaultParent (ImGuiIO* s, bool value){ s->ConfigViewportsNoDefaultParent=value;}
DIMGUI_EXPORT void DImGuiIOSetMouseDrawCursor (ImGuiIO* s, bool value){ s->MouseDrawCursor=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigMacOSXBehaviors (ImGuiIO* s, bool value){ s->ConfigMacOSXBehaviors=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigInputTrickleEventQueue (ImGuiIO* s, bool value){ s->ConfigInputTrickleEventQueue=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigInputTextCursorBlink (ImGuiIO* s, bool value){ s->ConfigInputTextCursorBlink=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigInputTextEnterKeepActive (ImGuiIO* s, bool value){ s->ConfigInputTextEnterKeepActive=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigDragClickToInputText (ImGuiIO* s, bool value){ s->ConfigDragClickToInputText=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigWindowsResizeFromEdges (ImGuiIO* s, bool value){ s->ConfigWindowsResizeFromEdges=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigWindowsMoveFromTitleBarOnly (ImGuiIO* s, bool value){ s->ConfigWindowsMoveFromTitleBarOnly=value;}
DIMGUI_EXPORT void DImGuiIOSetConfigMemoryCompactTimer (ImGuiIO* s, float value){ s->ConfigMemoryCompactTimer=value;}
DIMGUI_EXPORT void DImGuiIOSetWantCaptureMouse (ImGuiIO* s, bool value){ s->WantCaptureMouse=value;}
DIMGUI_EXPORT void DImGuiIOSetWantCaptureKeyboard (ImGuiIO* s, bool value){ s->WantCaptureKeyboard=value;}
DIMGUI_EXPORT void DImGuiIOSetWantTextInput (ImGuiIO* s, bool value){ s->WantTextInput=value;}
DIMGUI_EXPORT void DImGuiIOSetWantSetMousePos (ImGuiIO* s, bool value){ s->WantSetMousePos=value;}
DIMGUI_EXPORT void DImGuiIOSetWantSaveIniSettings (ImGuiIO* s, bool value){ s->WantSaveIniSettings=value;}
DIMGUI_EXPORT void DImGuiIOSetNavActive (ImGuiIO* s, bool value){ s->NavActive=value;}
DIMGUI_EXPORT void DImGuiIOSetNavVisible (ImGuiIO* s, bool value){ s->NavVisible=value;}
DIMGUI_EXPORT void DImGuiIOSetFramerate (ImGuiIO* s, float value){ s->Framerate=value;}
DIMGUI_EXPORT void DImGuiIOSetMetricsRenderVertices (ImGuiIO* s, int value){ s->MetricsRenderVertices=value;}
DIMGUI_EXPORT void DImGuiIOSetMetricsRenderIndices (ImGuiIO* s, int value){ s->MetricsRenderIndices=value;}
DIMGUI_EXPORT void DImGuiIOSetMetricsRenderWindows (ImGuiIO* s, int value){ s->MetricsRenderWindows=value;}
DIMGUI_EXPORT void DImGuiIOSetMetricsActiveWindows (ImGuiIO* s, int value){ s->MetricsActiveWindows=value;}
DIMGUI_EXPORT void DImGuiIOSetMetricsActiveAllocations (ImGuiIO* s, int value){ s->MetricsActiveAllocations=value;}
DIMGUI_EXPORT void DImGuiIOSetMouseDelta (ImGuiIO* s, ImVec2 value){ s->MouseDelta=value;}

// clang-format on